Softeq.XToolkit.WhiteLabel.Droid/ViewComponents/ViewModelComponent.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Developed by Softeq Development Corporation
+// http://www.softeq.com
+
+using System;
+using AndroidX.Fragment.App;
+
+namespace Softeq.XToolkit.WhiteLabel.Droid.ViewComponents
+{
+    public sealed class ViewModelComponent<TViewModel> where TViewModel : class
+    {
+        private TViewModel? _viewModel;
+
+        public ViewModelComponent(string viewModelKey)
+        {
+            ViewModelKey = viewModelKey;
+        }
+
+        public TViewModel ViewModel => _viewModel ?? throw new InvalidOperationException("ViewModel instance is not assigned");
```

Comment:
Should we really throw an exception here? Seems that we expect that _viewModel can become null when app is recreated

Reply:
Yes, we expect that at some point _viewModel might be null. But at the same time, we expect that ViewModel property will be used only after _viewModel restored. If we don't throw exception here when _viewModel is null - we will get NRE somewhere later either way

Reply:
And what if I want to check the property to know if I need to restore it?

Reply:
This seems like an internal logic of restoration - I could add a check inside RestoreViewModel method.

Reply:
Please add a comment to this property mentioning the exception

Reply:
@TimofeyBurak Could you please mention how to assign this property? (call initialize method)

Reply:
Documentation comments were added to the class